tests/unit/test_all_paths.cpp

```diff
@@ -0,0 +1,243 @@
+/*
+ * Copyright 2018-2019 Redis Labs Ltd. and Contributors
+ *
+ * This file is available under the Apache License, Version 2.0,
+ * modified with the Commons Clause restriction.
+ */
+
+#include "../../deps/googletest/include/gtest/gtest.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include "../../src/algorithms/algorithms.h"
+#include "../../src/util/arr.h"
```

Comment:
I think we need to move the function definitions in `arr.h` into a separate .c file before these tests will compile correctly for me/Circle.

Reply:
Already solved, simple casting.